ec/ecresource/snapshotrepositoryresource/schema.go

```diff
@@ -0,0 +1,196 @@
+		Optional:    true,
+		Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+			"region": {
+				Type:        types.StringType,
+				Description: "Allows specifying the signing region to use. Specifying this setting manually should not be necessary for most use cases. Generally, the SDK will correctly guess the signing region to use. It should be considered an expert level setting to support S3-compatible APIs that require v4 signatures and use a region other than the default us-east-1. Defaults to empty string which means that the SDK will try to automatically determine the correct signing region.",
+				Optional:    true,
+			},
+			"bucket": {
+				Type:        types.StringType,
+				Description: "(Required) Name of the S3 bucket to use for snapshots.",
+				Required:    true,
+			},
+			"access_key": {
+				Type:        types.StringType,
+				Description: "An S3 access key. If set, the secret_key setting must also be specified. If unset, the client will use the instance or container role instead.",
+				Optional:    true,
+			},
+			"secret_key": {
+				Type:        types.StringType,
+				Description: "An S3 secret key. If set, the access_key setting must also be specified.",
+				Optional:    true,
+				Sensitive:   true,
+			},
+			"server_side_encryption": {
+				Type:        types.BoolType,
+				Description: "When set to true files are encrypted on server side using AES256 algorithm. Defaults to false.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					planmodifier.DefaultValue(types.Bool{Value: false}),
+				},
+			},
+			"endpoint": {
+				Type:        types.StringType,
+				Description: "The S3 service endpoint to connect to. This defaults to s3.amazonaws.com but the AWS documentation lists alternative S3 endpoints. If you are using an S3-compatible service then you should set this to the service’s endpoint.",
+				Optional:    true,
+			},
+			"path_style_access": {
+				Type:        types.BoolType,
+				Description: "Whether to force the use of the path style access pattern. If true, the path style access pattern will be used. If false, the access pattern will be automatically determined by the AWS Java SDK (See AWS documentation for details). Defaults to false.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					planmodifier.DefaultValue(types.Bool{Value: false}),
+				},
+			},
+		}),
+	}
+}
+
+func genericSchema() tfsdk.Attribute {
+	return tfsdk.Attribute{
+		Description: "Generic repository settings.",
+		Optional:    true,
+		Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+			"type": {
+				Type:        types.StringType,
+				Description: "(Required) Repository type",
+				Required:    true,
+			},
+			"settings": {
+				Type:        types.StringType,
+				Description: "(Required) An arbitrary JSON object containing the repository settings.",
+				Required:    true,
```

Comment:
I think we'll end up wanting a plan modifier here to ignore diffs between equivalent json strings (e.g `{"a": 1, "b": 2}` and `{"b": 2, "a": 1}`). Happy to ignore than for now though.